File: src/main/java/com/example/Flight/model/Seat.java
package com.example.Flight.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Seat {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String seatNumber;
    private boolean isAvailable;

    @Enumerated(EnumType.STRING)
    private SeatType seatType;

    @Enumerated(EnumType.STRING)
    private SeatClass seatClass;

    @ManyToOne
    @JoinColumn(name = "flight_id")
    private Flight flight;
}

File: frontend/src/app/model/seat.ts
import { Airplane } from "./airplane";
import { ReservationSeats } from "./reservationSeat";

export interface Seat {
  id: number;
  seatNumber: string;
  seatType: string;
  seatClass: string;
  extraLegroom: boolean;
  exitRow: boolean;
}